src/web/src/pages/SubjectExplorer.vue

```diff
@@ -105,6 +106,11 @@ export default {
    * subjectCourseArr is an array of course objects
    */
   async created() {
+    const querySemester = this.$route.query.semester;
+    this.selectedSemester =
+      querySemester && querySemester != "null"
+        ? querySemester
+        : await getDefaultSemester();
     getCourses(this.selectedSemester).then((courses) => {
```

Comment:
would prefer this to use await syntax as well

`const courses = await getCourses(this.selectedSemester);`